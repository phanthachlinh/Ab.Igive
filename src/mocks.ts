import { ITranslationsDictionary } from "./types/general";

export const TEST_TRANSLATIONS: ITranslationsDictionary = {
	en: {
		homepage: {
			contactUsTitle: 'Still not sure what to do?',
			contactUsSubtitle: 'Get in touch. We’ll be happy to help you define your campaign and set up your fundraising page.',
			contactUs: 'CONTACT US'
		},
		eventPage: {
			title: 'Start your campaign today.',
			subTitle: 'It only takes two minutes to set up your fundraising page. Decide what to do and choose in which global campaign you want to get involved in. Pick a name, a photo and just like that, you’ll be ready to start raising money to help people who have been forced to flee.'
		},
		shared: {
			seeAllCampaigns: 'see all campaigns',
			searchForGlobalCamp: 'Search for a global campaign',
			copyLink: 'Copy link',
			donateNow: 'DONATE NOW',
			shareThisCampaign: ' Share this campaign',
			setupMyFundpage: 'Setup my fundraising page',
			shareThisFundPage: 'Share this fundraising page'
		}
	},
	hk: {
		homepage: {
			contactUsTitle: '還未肯定應該做甚麼？',
			contactUsSubtitle: '與我們聯繫吧。我們樂意幫你為籌款項目找到定位，以及協助你創建籌款專頁。',
			contactUs: '聯絡我們'
		},
		eventPage: {
			title: '今日就建立你的籌款專頁',
			subTitle: '創建你的籌款專頁只需兩分鐘。先決定你想開設的項目類型並選擇其中一個全球項目。為活動起一個名字以及挑選一張相片，就這麼簡單，你就準備好建立籌款專頁，協助被迫逃離家園的難民。'
		},
		shared: {
			seeAllCampaigns: '查看所有籌款活動',
			searchForGlobalCamp: '搜尋全球項目',
			copyLink: '複製連結',
			donateNow: '立即捐款',
			shareThisCampaign: '分享此籌款活動',
			setupMyFundpage: '設置我的籌款頁面',
			shareThisFundPage: '分享此籌款專頁'
		}
	}
}
