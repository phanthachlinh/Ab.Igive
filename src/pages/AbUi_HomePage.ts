/**
* Class renders change on the home page (/)
**/
import { ITranslationsDictionary } from "../types/general"
export default class AbUi_HomePage {

	translations: ITranslationsDictionary
	pageLanguage: string

	constructor(translations: ITranslationsDictionary, pageLanguage: string) {

		this.translations = translations
		this.pageLanguage = pageLanguage

	}


	render() {

	}

}
