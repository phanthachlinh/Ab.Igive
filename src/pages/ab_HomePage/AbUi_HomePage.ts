/**
* Class renders change on the home page (/)
**/
import { ITranslationsDictionary } from "../../types/general"
export default class AbUi_HomePage {

	eventsThumbnailContainer: HTMLUListElement | null
	translations: ITranslationsDictionary
	pageLanguage: string

	constructor(translations: ITranslationsDictionary, pageLanguage: string) {

		this.translations = translations
		this.pageLanguage = pageLanguage
		this.eventsThumbnailContainer = document.querySelector('.examples-wrapper-inner ul')
	}

	/**
	* renders contact us element into last position in the list of projects
	**/
	private _renderContactUsThumbnail(): void {

		if (this.eventsThumbnailContainer) {

			let contactUsComponent =
				`<section class="content-contact-us">
					<div class="content-contact-us-inner">
						<h2>${this.translations[this.pageLanguage].homepage.contactUsTitle}</h2>
						<p>${this.translations[this.pageLanguage].homepage.contactUsSubtitle}</p>
						<a class="custom__button-main button-blue-border" href="${window.location.origin}/contact_forms/new">
							${this.translations[this.pageLanguage].homepage.contactUs}
							</a>
					</div>
				</section>`
			this.eventsThumbnailContainer.insertAdjacentHTML('beforeend', contactUsComponent)
		} else
			console.error('Event thumbnail container not found')
	}

	render() {

		this._renderContactUsThumbnail()
	}

}
