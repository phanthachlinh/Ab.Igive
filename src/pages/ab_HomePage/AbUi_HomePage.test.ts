import { JSDOM } from "jsdom"
import { TEST_TRANSLATIONS } from "../../mocks"
import AbUi_HomePage from "./AbUi_HomePage"
declare var global: any
let abUiHomePageObj: any
console.error = jest.fn((errmsg: string) => {
	throw errmsg
})
describe('AbUi_HomePage', () => {
	describe('_renderContactUsThumbnail', () => {
		it('should render contact us thumbnail', () => {
			global.document.documentElement.innerHTML = new JSDOM(``)
			abUiHomePageObj = new AbUi_HomePage(TEST_TRANSLATIONS, 'en')
			expect(() => { abUiHomePageObj._renderContactUsThumbnail() }).toThrow('Event thumbnail container not found')
		})
		it('should match snapshot', () => {
			global.document.documentElement.innerHTML = new JSDOM(`
        <div class="examples-wrapper-inner">
          <ul>
          </ul>
        </div>
        `).serialize()
			abUiHomePageObj = new AbUi_HomePage(TEST_TRANSLATIONS, 'en')
			abUiHomePageObj._renderContactUsThumbnail()
			expect(document.documentElement.innerHTML).toMatchSnapshot()
		})
	})

})
/*
global.document.documentElement.innerHTML = new JSDOM(`
  <div class="examples-wrapper-inner">
    <ul>
    </ul>
  </div>
  `)
*/
