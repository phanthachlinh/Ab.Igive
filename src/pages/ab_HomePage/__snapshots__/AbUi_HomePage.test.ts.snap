// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbUi_HomePage __renderContactUsThumbnail should match snapshot 1`] = `
"<head></head><body><div class=\\"examples-wrapper-inner\\">
          <ul>
          <section class=\\"content-contact-us\\">
					<div class=\\"content-contact-us-inner\\">
						<h2>Still not sure what to do?</h2>
						<p>Get in touch. Weâ€™ll be happy to help you define your campaign and set up your fundraising page.</p>
						<a class=\\"custom__button-main button-blue-border\\" href=\\"http://localhost/contact_forms/new\\">
							CONTACT US
							</a>
					</div>
				</section></ul>
        </div>
        </body>"
`;
